common/lib/dependabot/dependency_updaters/README.md

```diff
@@ -0,0 +1,72 @@
+# Dependency Updater
+
+A *dependency updater* is responsible for updating a dependent package
+according to a set of requirements.
+It provides a minimal interface for languages to add Dependabot support
+by delegating responsibilities to a package manager for that language.
+
+## Public API
+
+Each `Dependabot::DependencyUpdaters` class inherits from `Base` 
+and implements the following methods:
+
+### `#update(dependency:, requirements:)`
+
+Updates a dependency according to the provided requirements.
+This method is called within the root directory of a project.
+
+#### Arguments
+
+| Argument        | Type                               | Description                                      |
+| --------------- | ---------------------------------- | ------------------------------------------------ |
+| `dependency`    | `Dependabot::Dependency`           | The package dependency to update                 |
+| `requirements`  | `VersionRange`                     | The updated version constraints                  |
+| `configuration` | `Dependabot::Config::UpdateConfig` | The configuration options for performing updates |
+
+#### Return
+
+Returns `true` if the dependency was successfully updated 
+to satisfy the new requirements.
+
+#### Raise
+
+This method should raise an exception to communicate underlying errors
```

Comment:
Is there a case where `#update` would return false and not raise an error? In our current implementation the specific errors returned are pretty important and can dictate how we present the error to the user, whether we should attempt to retry later or could trigger alerts. We probably need to provide some kinda of guidance on what errors to raise so we don't need to rely on parsing error strings?